language/move-lang/src/unit_test/filter_test_members.rs

```diff
@@ -291,18 +296,25 @@ fn filter_tests_from_module_member(
         PM::Constant(constant) => &constant.attributes,
     };
 
-    if should_remove_node(context.env, attrs) {
+    if should_remove_node(context.env, attrs, is_lib) {
         None
     } else {
         Some(module_member)
     }
 }
 
 // A module member should be removed if:
-// * It is annotated as a test function (test_only, test, abort); and
-// * Test mode is not set
-fn should_remove_node(env: &CompilationEnv, attrs: &[P::Attributes]) -> bool {
+// * It is annotated as a test function (test_only, test, abort) and test mode is not set; or
+// * If it is a library and is annotated as #[test]
+fn should_remove_node(env: &CompilationEnv, attrs: &[P::Attributes], is_lib: bool) -> bool {
     attrs.iter().flat_map(test_attributes).count() > 0 && !env.flags().is_testing()
+        || (is_lib
+            && attrs.iter().any(|attrs| {
```

Comment:
iter().flatten().any()? not sure if that works but what is here is fine

Reply:
also a thought, there is already this pass above with the flat_map to filter the test_attributes. Maybe we should just collect those in a vec or something and then filter them again here

Reply:
Tried the flatten and Rust wasn't happy with me. But I'll re-use the results from the flat_map from above here.

Reply:
These are just thoughts, what you had was clear just felt a bit awkward. If the other thing is more awkward, the original is fine

Reply:
Just made the change to `flat_map` -- actually works out a lot nicer than what is here currently since `test_attributes` will parse into a TestingAttribute as well.